core/notification/router_receiver_service.go

```diff
@@ -51,10 +84,55 @@ func (s *RouterReceiverService) PrepareMetaMessages(ctx context.Context, n Notif
 		})
 	}
 
-	var metaMessagesNum = len(metaMessages)
-	if metaMessagesNum > s.deps.Cfg.MaxMessagesReceiverFlow {
-		return nil, nil, errors.ErrInvalid.WithMsgf("sending %d messages exceed max messages receiver flow threshold %d. this will spam and broadcast to %d channel. found %d receiver selectors passed, you might want to check your receiver selectors configuration", metaMessagesNum, s.deps.Cfg.MaxMessagesReceiverFlow, metaMessagesNum, len(n.ReceiverSelectors))
+	return metaMessages, notificationLogs, nil
+}
+
+func (s *RouterReceiverService) handleConfigCase(ctx context.Context, selectors []map[string]any) ([]receiver.Receiver, map[uint64]map[string]interface{}, error) {
+	var receiverIDs []uint64
+	userConfigs := make(map[uint64]map[string]interface{})
+
+	for _, selector := range selectors {
+		if idStr, ok := selector["id"].(string); ok {
+			id, err := strconv.ParseUint(idStr, 10, 64)
+			if err != nil {
+				return nil, nil, errors.ErrInvalid.WithMsgf("invalid receiver id: %s", idStr)
+			}
+			receiverIDs = append(receiverIDs, id)
+		}
+		if config, ok := selector["config"].(map[string]interface{}); ok {
+			for _, id := range receiverIDs {
+				if userConfigs[id] == nil {
+					userConfigs[id] = make(map[string]interface{})
+				}
+				for k, v := range config {
+					userConfigs[id][k] = v
+				}
+			}
+		}
 	}
 
-	return metaMessages, notificationLogs, nil
+	rcvs := make([]receiver.Receiver, 0, len(receiverIDs))
+	for _, id := range receiverIDs {
+		rcv, err := s.deps.ReceiverService.Get(ctx, id)
+		if err != nil {
+			return nil, nil, err
+		}
+		rcvs = append(rcvs, *rcv)
+	}
+
+	return rcvs, userConfigs, nil
+}
+
+func (s *RouterReceiverService) handleNonConfigCase(ctx context.Context, selectors []map[string]any) ([]receiver.Receiver, error) {
```

Comment:
I dont think we need to do case by case with if like this.

How about to make it less complicated, we do like this for now
config overriding mechanism should only work if it is only 1 receiver selector and it is using id as part of the labels otherwise it will be ignored.

Even after receiver evaluation, if turns out the receiver is not a parent receiver (means config already exist), we ignore the overriden config.